tests/unit/test_cli.py

```diff
@@ -361,3 +363,21 @@ def test_validate_group_no_ucx(mocker, caplog):
     mocker.patch("databricks.labs.ucx.installer.InstallationManager.for_user", return_value=None)
     validate_groups_membership(w)
     assert "Couldn't find UCX configuration" in caplog.messages[0]
+
+
+def test_save_aws_iam_profiles_no_profile(mocker, caplog):
+    w = create_autospec(WorkspaceClient)
+    w.current_user.me = lambda: iam.User(user_name="test_user", groups=[iam.ComplexValue(display="admins")])
+    mocker.patch("databricks.labs.ucx.installer.InstallationManager.for_user", return_value=None)
+    os.environ["AWS_DEFAULT_PROFILE"] = ""
+    save_aws_iam_profiles(w)
+    assert "AWS Profile is not specified." in caplog.messages[0]
+
+
+def test_save_aws_iam_profiles_no_cli(mocker, caplog):
+    w = create_autospec(WorkspaceClient)
+    w.current_user.me = lambda: iam.User(user_name="test_user", groups=[iam.ComplexValue(display="admins")])
+    mocker.patch("databricks.labs.ucx.assessment.aws.AWSResources.validate_connection", return_value=False)
```

Comment:
Red flag: monkey patching the code we wrote. This is means incorrect/substandard implementation. This is not allowed 

Also:

Why do you need to validate a connection? Listing roles/whatever  will tell you that connection is invalid when it's invalid. Remove this method, it's redundant.

Reply:
I would like to perform a "pre check" before I bombard with multithreaded requests.

Reply:
Fixed